Refactoring for Intelligent Volume Provisioning

aravindavk opened the pull request:
Added more device utils and `fstab` handling library

Updates: #466
Signed-off-by: Aravinda VK <avishwan@redhat.com>

Madhu-1 commented on glusterd2/volume/volume-utils.go:
```diff
@@ -17,6 +17,21 @@ import (
+func GetRedundancy(disperse uint) int {
+	var temp, l, mask uint
+	temp = disperse
+	l = 0
```
no need to set it explicitly to zero

aravindavk replied:
This func is moved from `volume-create-disperse.go` to utils. Implementation not changed

aravindavk replied:
fixed.

aravindavk commented:
Note for reviewers:

- Moved `glusterd2/commands/volumes/events.go` to `glusterd2/volume/events.go`
- Moved `getRedundancy` function from `glusterd2/commands/volumes/volume-create-disperse.go` to `glusterd2/volume/volume-utils.go` file
- Moved  `glusterd2/commands/volumes/brickutils.go` to `pkg/api/brickutils.go`
- Added three additional fields to Device Resp

         + VgName        string `json:"vg-name"`
         + AvailableSize uint64 `json:"available-size"`
         + Used          bool   `json:"used"`
- New file added for `fstab` functionalities ` plugins/device/deviceutils/fstab.go`
- More functions added to ` plugins/device/deviceutils/utils.go` 
- Updated `AvailableSize` during device add

prashanthpai commented on pkg/api/brickutils.go:
```diff
@@ -1,14 +1,13 @@
 
-func nodesFromVolumeCreateReq(req *api.VolCreateReq) ([]uuid.UUID, error) {
+// NodesFromVolumeCreateReq extracts the list of nodes from Volume Create request
+func NodesFromVolumeCreateReq(req *VolCreateReq) ([]uuid.UUID, error) {
```
You can make this as a method of VolCreateReq structure and then use it as `req.Nodes()`

prashanthpai commented on pkg/api/brickutils.go:
```diff
@@ -26,7 +25,8 @@ func nodesFromVolumeCreateReq(req *api.VolCreateReq) ([]uuid.UUID, error) {
 
-func nodesFromVolumeExpandReq(req *api.VolExpandReq) ([]uuid.UUID, error) {
+// NodesFromVolumeExpandReq extracts list of Peer IDs from Volume Expand request
+func NodesFromVolumeExpandReq(req *VolExpandReq) ([]uuid.UUID, error) {
```
Same here.

aravindavk commented:
@prashanthpai @Madhu-1 addressed review comments, please review

prashanthpai commented on plugins/device/deviceutils/fstab.go:
```diff
@@ -0,0 +1,173 @@
+	if err != nil {
+		return nil, err
+	}
+	defer lockfile.Close()
```
The file is closed when the function returns. Doesn't closing file description implicitly unlock ?

aravindavk replied:
yes, will remove this

prashanthpai commented:
retest this please

prashanthpai commented:
retest this please
